Sources/Datadog/RUMMonitor.swift

```diff
@@ -283,14 +283,19 @@ public class RUMMonitor: DDRUMMonitor, RUMCommandSubscriber {
     override public func addError(
         message: String,
         source: RUMErrorSource,
+        stack: String?,
         attributes: [AttributeKey: AttributeValue],
         file: StaticString?,
         line: UInt?
     ) {
-        var stack: String? = nil
-        if let file = file, let fileName = "\(file)".split(separator: "/").last, let line = line {
-            stack = "\(fileName):\(line)"
-        }
+        let stack: String? = stack ?? {
+            if let file = file,
+               let fileName = "\(file)".split(separator: "/").last,
+               let line = line {
+                return "\(fileName):\(line)"
+            }
+            return nil
+        }()
         addError(message: message, stack: stack, source: RUMInternalErrorSource(source), attributes: attributes)
```

Comment:
The:
```swift
internal func addError(
   message: String,
   stack: String?,
   source: RUMInternalErrorSource,
   attributes: [AttributeKey: AttributeValue]
)
```
is not needed anymore. It was there to provide explicit internal API for passing `stack` parameter. Now we can replace it's only usage with:
```swift
override public func addError(
   message: String,
   source: RUMErrorSource,
   stack: String?,
   attributes: [AttributeKey: AttributeValue],
   file: StaticString?,
   line: UInt?
)
```
(it's only called once in `RUMErrorsIntegration`).

Reply:
it seems it was there to receive `RUMInternalErrorSource`
i guess we should keep it as we can't make `RUMInternalErrorSource` public

Reply:
Ah, indeed 👍